segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/GetInfoOperation.java

```diff
@@ -62,7 +62,8 @@ public SegmentProperties call() throws IOException {
                 throw fnf;
             }
 
-            result = new StreamSegmentInformation(segmentName, length, isSealed, false, new ImmutableDate());
+            result = new StreamSegmentInformation(segmentName, length, isSealed, false,
+                    new ImmutableDate(findAllRaw(segmentName)[0].getModificationTime()));
```

Comment:
If there is more than one, why does it take the first? Should it be the last?
Also I'm not sure this is even what we want. This is the file modification time. We're using this to compare versions of the info to determine which file is more up-to-date. It's possible for something to for example modify the metadata or acls on a file an alter the modification time. We probably want some sort of sequence or number supplied by SSS to determine which is more recent.

Reply:
This needs fixing in case of HDFS implementation. For this limited case, as statefile is created and updated frequently and is deleted after that, this will work.
Looking at how we can use sequence number etc to fix this.